signing-jobs/build/sign-artifacts/umb_producer.py

```diff
@@ -0,0 +1,324 @@
+                       verify=ssl.get_default_verify_paths().openssl_cafile)
+    if res.status_code == 200:
+        return res.json()
+    else:
+        raise(Exception(res.reason))
+
+
+def presend_validation(message):
+    """Verify the message we want send over the bus has all the required
+fields"""
+    for field in SIGN_REQUEST_MESSAGE_FIELDS:
+        if field not in message:
+            return field
+    return True
+
+
+def oc_image_info(pullspec):
+    """Get metadata for an image at the given `pullspec`
+
+    :return: a dict with the serialzed JSON from the 'oc image info'
+call
+    """
+    image_info_raw = subprocess.check_output(
+        ['oc', 'image', 'info', '-o', 'json', pullspec])
+    return json.loads(image_info_raw)
+
+
+def get_bus_producer(env, certificate, private_key, trusted_certificates):
+    """This is just a wrapper around creating a producer. We're going to
+do need this in multiple places though, so we want to ensure we do it
+the same way each time.
+    """
+    return AMQProducer(urls=URLS[env or 'stage'],
+                       certificate=certificate,
+                       private_key=private_key,
+                       trusted_certificates=trusted_certificates,
+                       topic=TOPIC)
+
+
+######################################################################
+@cli.command("message-digest", short_help="Sign a sha256sum.txt file")
+@requestor
+@product
+@request_id
+@sig_keyname
+@release_name
+@client_cert
+@client_key
+@env
+@noop
+@ca_certs
+@click.pass_context
+def message_digest(ctx, requestor, product, request_id, sig_keyname,
+                   release_name, client_cert, client_key, env, noop,
+                   ca_certs):
+    """Sign a 'message digest'. These are sha256sum.txt files produced by
+the 'sha256sum` command (hence the strange command name). In the ART
+world, this is for signing message digests from extracting OpenShift
+tools, as well as RHCOS bare-betal message digests.
+    """
+    message = {
+        "artifact": get_digest_base64(MESSAGE_DIGESTS[product].format(
+            release_name=release_name)),
+        "artifact_meta": {
+            "product": product,
+            "release_name": release_name,
+            "name": "sha256sum.txt.sig",
+            "type": "message-digest",
+        },
+        "request_id": request_id,
+        "requestor": requestor,
+        "sig_keyname": sig_keyname,
+    }
+
+    validated = presend_validation(message)
+    if validated is True:
+        print("Message contains all required fields")
+        to_send = json.dumps(message)
+    else:
+        print("Message missing required field: {}".format(validated))
+        exit(1)
+
+    if noop:
+        print("Message we would have sent over the bus:")
+        print(to_send)
+    else:
+        producer = get_bus_producer(env, client_cert, client_key, ca_certs)
+        producer.send_msg({'tester': requestor, 'to': 'test_msg'}, to_send)
```

Comment:
I just noticed this (also in `json-digest`). I'll ask for suggestions on what to put there instead.

Reply:
Figured that out.